### ISSUE TITLE
Importing module is undefined in development

### ISSUE THREAD

**Issue by Nick-Mazuk** [sveltejs/kit#737]
**Describe the bug**

Importing a npm package returns undefined in development but not in production.

```
<script>
    import marked from 'marked'
    // marked is undefined
</script>
```

This bug does not seem to be isolated to `marked`. I've encountered it with `markdown-it`, `snarkdown`, and several of my own npm packages.

**Logs**

```
Cannot read property 'setOptions' of undefined

TypeError: Cannot read property 'setOptions' of undefined
    at index.svelte:5:21
    at Object.$$render (/path/to/repo/node_modules/.pnpm/svelte@3.35.0/node_modules/svelte/internal/index.js:1369:22)
    at Object.default (root.svelte:46:46)
    at /path/to/repo/.svelte/dev/components/layout.svelte:5:41
    at Object.$$render (/path/to/repo/node_modules/.pnpm/svelte@3.35.0/node_modules/svelte/internal/index.js:1369:22)
    at root.svelte:42:13
    at $$render (/path/to/repo/node_modules/.pnpm/svelte@3.35.0/node_modules/svelte/internal/index.js:1369:22)
    at Object.render (/path/to/repo/node_modules/.pnpm/svelte@3.35.0/node_modules/svelte/internal/index.js:1377:26)
    at get_response (file:///path/to/repo/node_modules/.pnpm/@sveltejs/kit@1.0.0-next.61_svelte@3.35.0+vite@2.1.3/node_modules/@sveltejs/kit/dist/ssr.js:2360:27)
    at async render_page (file:///path/to/repo/node_modules/.pnpm/@sveltejs/kit@1.0.0-next.61_svelte@3.35.0+vite@2.1.3/node_modules/@sveltejs/kit/dist/ssr.js:2472:19)
```

**To Reproduce**

Minimal reproduction (created from steps below): https://github.com/Nick-Mazuk/sveltekit-import

```
pnpm init svelte@next
pnpm i
pnpm i marked
```

Then, in the `src/routes/index.svelte` file, just add this to the `<script>` tag:

```js
import marked from 'marked'

marked.setOptions({
    smartypants: true,
})
```

```
pnpm dev
```

Then, you'll get the error when you visit the site.

_However…_

If you build the project then start the production server, everything works correctly.

```
pnpm build
pnpm start
```

**Expected behavior**
No error in development. Should get same results as in production.

**Stacktraces**
No stack trace.

**Information about your SvelteKit Installation:**

```
System:
    OS: macOS 11.2
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 122.65 MB / 64.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.17.0 - ~/.nvm/versions/node/v12.17.0/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.6.0 - ~/.nvm/versions/node/v12.17.0/bin/npm
  Browsers:
    Brave Browser: 89.1.22.70
    Firefox: 84.0.2
    Safari: 14.0.3
  npmPackages:
    @sveltejs/kit: next => 1.0.0-next.61 
    svelte: ^3.29.0 => 3.35.0 
```

Same results on node and Vercel adapters.

**Severity**
Sever. Completely blocking development.

**Comment by Rich-Harris** [comment]
Well this is weird, to say the least. I could reproduce it in a fresh project, but not in an example app in this monorepo or in a test (#747). And I'm using imports from node_modules in a production app.

I can only imagine something about CommonJS conversion changed recently in Vite, or esbuild, or something? But I can't seem to figure out what exactly

**Comment by Nick-Mazuk** [comment]
Since you can't replicate it in this monorepo but you can with new projects, I'm assuming the bug is from a dependency update. Though even weirder, I don't think it's a direct update to Vite or ESBuild—but perhaps one of their dependencies.

To debug, I tried using the same versions for several dependencies as in the svelte/kit monorepo. Did this by manually updating both the package.json and pnpm-lock.yaml. These are the dependencies I tried:

- Vite
- ESBuild
- Svelte
- cheap-watch
- sade
- rollup
- resolve
- postcss

In short, if it is a dependency issue, it's not going to be obvious.

Going to see if I can reproduce this with Vite itself.

Note: with more recent versions of ESBuild, there are breaking changes that affect CommonJS modules. (https://github.com/evanw/esbuild/releases/tag/v0.10.0, https://github.com/evanw/esbuild/releases/tag/v0.11.0). However, Vite does not support v0.10.0, and my pnpm-lock.yaml uses 0.9.7. So this should not be the issue.

**Comment by stefandesu** [comment]
Not sure if this helps, but after removing `noExternal: Object.keys(pkg.dependencies || {})` from `svelte.config.cjs`, it suddenly works.

**Comment by n4bb12** [comment]
It might be the same issue but I'm not sure.

During development
```
import pc from "@prisma/client"
// Cannot read property 'PrismaClient' of undefined
```
```
import * as pc from "@prisma/client"
// ✔
```

During production the opposite works
```
import pc from "@prisma/client"
// ✔
```
```
import * as pc from "@prisma/client"
// Cannot read property 'PrismaClient' of undefined
```

Currently working around this by keeping an uncommitted local change.

**Comment by stefandesu** [comment]
@n4bb12 I ended up doing something similar, except making it dependent on `NODE_ENV` and therefore being able to commit it:

```js
{
  vite: {
    ssr: {
      noExternal: process.env.NODE_ENV === "production" ? Object.keys(pkg.dependencies || {}) : [],
    },
  }
}
```

**Comment by n4bb12** [comment]
> @n4bb12 I ended up doing something similar, except making it dependent on `NODE_ENV` and therefore being able to commit it:

Thanks but that's unfortunately not fixing `Cannot read property 'PrismaClient' of undefined`. Feel free to try it out yourself.

**Comment by Sebastiony-Uni** [comment]
This issue has just occured for me importing a smaller package called Steps https://github.com/shaozi/svelte-steps. When importing through the recommended means `import { Steps } from 'svelte-steps';` I'd use the component in the html and get the error

![image](https://user-images.githubusercontent.com/69179091/211528753-da209e30-5d81-43c8-acdd-b432a8970dc7.png)

managed to fix this issue by doing what you stated above (`import * as Steps from 'svelte-steps';`) then the component in the html is `<Steps.Steps />`

An interesting note is that if I commented out the component in the code and loaded the webpage, then uncommented and saved it, the component would work. It wasn't until I refreshed the page that it would crash again. I assume its because its reimporting on refresh.

Sorry I don't have a reproducible example at this stage but thought It would be good to at least note the fact that this issue is still occuring
